google-cloud-examples/src/main/java/com/google/cloud/examples/storage/snippets/StorageSnippets.java

```diff
@@ -1577,4 +1582,140 @@ public URL generateV4GPutbjectSignedUrl(String bucketName, String objectName)
     // [END storage_generate_upload_signed_url_v4]
     return url;
   }
+
+  /** Example of creating an HMAC key for a service account * */
+  public HmacKey createHmacKey(String serviceAccountEmail) throws StorageException {
+    // [START storage_create_hmac_key]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The service account email for which the new HMAC key will be created.
+    // String serviceAccountEmail = "service-account@iam.gserviceaccount.com";
+    ServiceAccount account = ServiceAccount.of(serviceAccountEmail);
+    HmacKey hmacKey = storage.createHmacKey(account);
+
+    String secret = hmacKey.getSecretKey();
+    HmacKeyMetadata metadata = hmacKey.getMetadata();
+
+    System.out.println("The base64 encoded secret is: " + secret);
+    System.out.println("Do not miss that secret, there is no API to recover it.");
+    System.out.println("The HMAC key metadata is:");
+    System.out.println("Id: " + metadata.getId());
+    System.out.println("AccessId: " + metadata.getAccessId());
+    System.out.println("ProjectId: " + metadata.getProjectId());
+    System.out.println("ServiceAccountEmail: " + metadata.getServiceAccount().getEmail());
+    System.out.println("State: " + metadata.getState().toString());
+    System.out.println("TimeCreated: " + new Date(metadata.getCreateTime()).toString());
+    System.out.println("Updated: " + new Date(metadata.getUpdateTime()).toString());
+    System.out.println("Etag: " + metadata.getEtag());
+    // [END storage_create_hmac_key]
+    return hmacKey;
+  }
+
+  /** Example of retrieving the metadata of an existing HMAC key. * */
+  public HmacKeyMetadata getHmacKey(String accessId) throws StorageException {
+    // [START storage_get_hmac_key]
+    // Instantiate a Google Cloud Storage client
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+
+    // The access ID of the HMAC key, e.g. "GOOG0234230X00"
+    // String accessId = "GOOG0234230X00";
```

Comment:
Why is this commented out?

Reply:
The code section between the START and END region tags are extracted and put on the doc site. When copy and pasting the sample into their own project, the developer is expected to uncomment the variable declaration and replace the value with their own.